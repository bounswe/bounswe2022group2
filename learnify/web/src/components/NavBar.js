import logo from '../images/logo-dblue.png' 
import bell from '../images/notification-icon.svg'
import React, {useState, useEffect } from 'react'
import './component_styles.css';
import axios from 'axios'

export default function NavBar(){

    const [searchResults, setSearchResults] = useState([])

    const [searchInput, setSearchInput] = useState("");

    const handleChange = (e) => {
        e.preventDefault();
        setSearchInput(e.target.value);
    }

    useEffect(() => {
        const getSearchResults = async () => {
            const res = await axios.get(`${process.env.REACT_APP_BACKEND_BASE_URL}learningspace?query=${searchInput}`)
            setSearchResults(res.data.learning_spaces)
        }
        getSearchResults()
    }, [searchInput])

    console.log(searchResults)

    return <nav className="nav" data-testid="navbar">
        <a href="/home"><img src={logo} alt="Learnify Logo" height={60} /></a>
        <ul>
            
                <div className="relative">
                    <input className='search-input-field' type="text" placeholder="Search learning spaces"  size={30} onChange={handleChange} value={searchInput}/>

                    <div className="search-results-container">
                        {searchResults.map(result => (
                            <ul>
                                <li className='search-result-item'><a href={`/learningspace/${result.id}`}>{result.title}</a></li>
                            </ul>
                        ))}
                    </div>
                </div>
            
            <li>
                <a href="/notifications" className='navBarText'><img src={bell} alt="Learnify Logo" height={20} /></a>
            </li>
            <li>
                <a href="/home" className='navBarText'>Home</a>
            </li>
            <li>
                <a href="/categories" className='navBarText'>Categories</a>
            </li>
            <li>
                <a href="/about" className='navBarText'>About</a>
            </li>
            <li>
                <a href="/contact" className='navBarText'>Contact</a>
            </li>
            <li>
                <a href="/profile" className='navBarText'>Profile</a>
            </li>
        </ul>
    </nav>

}