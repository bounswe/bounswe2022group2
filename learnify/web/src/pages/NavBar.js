import logo from '../images/logo-dblue.png' 
import bell from '../images/notification-icon.svg'
export default function NavBar(){
    return <nav className="nav">
        <a href="/"><img src={logo} alt="Learnify Logo" height={60} /></a>
        <ul>
            
                <label className="navSearch"><input type="text" placeholder="Search.."  size={30}/> </label>
            
            <li>
                <a href="/home" className='navBarText'><img src={bell} alt="Learnify Logo" height={30} /></a>
            </li>
            <li>
                <a href="/home" className='navBarText'>Home</a>
            </li>
            <li>
                <a href="/categories" className='navBarText'>Categories</a>
            </li>
            <li>
                <a href="/about" className='navBarText'>About</a>
            </li>
            <li>
                <a href="/contact" className='navBarText'>Contact</a>
            </li>
            <li>
                <a href="/profile" className='navBarText'>Profile</a>
            </li>
        </ul>
    </nav>

}