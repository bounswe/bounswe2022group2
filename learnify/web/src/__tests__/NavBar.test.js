import React from "react";
import NavBar from "./../components/NavBar";
import { screen, render, fireEvent } from "@testing-library/react";
import renderer from 'react-test-renderer';
import { BrowserRouter } from "react-router-dom";

describe("NavBar", () => {
    test('should render component layout correctly', async () => {
        const component = renderer.create(<BrowserRouter><NavBar /></BrowserRouter>).toJSON();
        expect(component).toMatchSnapshot();
    });
}   );

test('navbar renders correctly', async () => {
    render(<BrowserRouter><NavBar /></BrowserRouter>);
    const navbar = screen.getByTestId(/navbar/i);
    expect(navbar).toBeInTheDocument();
}   );

test('navbar logo renders correctly', async () => {
    render(<BrowserRouter><NavBar /></BrowserRouter>);
    const logo = screen.getByAltText(/Learnify Logo Navbar/i);
    expect(logo).toBeInTheDocument();
}   );

test('navbar search input renders correctly', async () => {
    render(<BrowserRouter><NavBar /></BrowserRouter>);
    const searchInput = screen.getByPlaceholderText(/Search learning spaces/i);
    expect(searchInput).toBeInTheDocument();
}   );

test('navbar search bar should accept input', async () => {
    render(<BrowserRouter><NavBar /></BrowserRouter>);
    const searchInput = screen.getByPlaceholderText(/Search learning spaces/i);
    fireEvent.change(searchInput, { target: { value: 'test' } });
    expect(searchInput.value).toBe('test');
}   );

test('navbar notifications icon renders correctly', async () => {
    render(<BrowserRouter><NavBar /></BrowserRouter>);
    const notificationIcon = screen.getByAltText(/Notifications Icon/i);
    expect(notificationIcon).toBeInTheDocument();
}   );

test('navbar notifications icon should be clickable', async () => {
    render(<BrowserRouter><NavBar /></BrowserRouter>);
    const notificationIcon = screen.getByAltText(/Notifications Icon/i);
    fireEvent.click(notificationIcon);
    expect(notificationIcon).toBeInTheDocument();
}   );

test('navbar home button renders correctly', async () => {
    render(<BrowserRouter><NavBar /></BrowserRouter>);
    const homeButton = screen.getByText(/Home/i);
    expect(homeButton).toBeInTheDocument();
}   );

test('navbar home button should be clickable', async () => {
    render(<BrowserRouter><NavBar /></BrowserRouter>);
    const homeButton = screen.getByText(/Home/i);
    fireEvent.click(homeButton);
    expect(homeButton).toBeInTheDocument();
}   );

test('navbar home button href should be correct', async () => {
    render(<BrowserRouter><NavBar /></BrowserRouter>);
    const homeButton = screen.getByText(/Home/i);
    expect(homeButton.href).toBe('http://localhost/home');
}   );

test('navbar categories button renders correctly', async () => {
    render(<BrowserRouter><NavBar /></BrowserRouter>);
    const categoriesButton = screen.getByText(/Categories/i);
    expect(categoriesButton).toBeInTheDocument();
}   );

test('navbar categories button should be clickable', async () => {
    render(<BrowserRouter><NavBar /></BrowserRouter>);
    const categoriesButton = screen.getByText(/Categories/i);
    fireEvent.click(categoriesButton);
    expect(categoriesButton).toBeInTheDocument();
}   );

test('navbar categories button href should be correct', async () => {
    render(<BrowserRouter><NavBar /></BrowserRouter>);
    const categoriesButton = screen.getByText(/Categories/i);
    expect(categoriesButton.href).toBe('http://localhost/categories');
}   );

test('navbar about button renders correctly', async () => {
    render(<BrowserRouter><NavBar /></BrowserRouter>);
    const aboutButton = screen.getByText(/About/i);
    expect(aboutButton).toBeInTheDocument();
}   );

test('navbar about button should be clickable', async () => {
    render(<BrowserRouter><NavBar /></BrowserRouter>);
    const aboutButton = screen.getByText(/About/i);
    fireEvent.click(aboutButton);
    expect(aboutButton).toBeInTheDocument();
}   );

test('navbar about button href should be correct', async () => {
    render(<BrowserRouter><NavBar /></BrowserRouter>);
    const aboutButton = screen.getByText(/About/i);
    expect(aboutButton.href).toBe('http://localhost/about');
}   );

test('navbar contact button renders correctly', async () => {
    render(<BrowserRouter><NavBar /></BrowserRouter>);
    const contactButton = screen.getByText(/Contact/i);
    expect(contactButton).toBeInTheDocument();
}   );

test('navbar contact button should be clickable', async () => {
    render(<BrowserRouter><NavBar /></BrowserRouter>);
    const contactButton = screen.getByText(/Contact/i);
    fireEvent.click(contactButton);
    expect(contactButton).toBeInTheDocument();
}   );

test('navbar contact button href should be correct', async () => {
    render(<BrowserRouter><NavBar /></BrowserRouter>);
    const contactButton = screen.getByText(/Contact/i);
    expect(contactButton.href).toBe('http://localhost/contact');
}   );

test('navbar profile button renders correctly', async () => {
    render(<BrowserRouter><NavBar /></BrowserRouter>);
    const profileButton = screen.getByText(/Profile/i);
    expect(profileButton).toBeInTheDocument();
}   );

test('navbar profile button should be clickable', async () => {
    render(<BrowserRouter><NavBar /></BrowserRouter>);
    const profileButton = screen.getByText(/Profile/i);
    fireEvent.click(profileButton);
    expect(profileButton).toBeInTheDocument();
}   );

test('navbar profile button href should be correct', async () => {
    render(<BrowserRouter><NavBar /></BrowserRouter>);
    const profileButton = screen.getByText(/Profile/i);
    expect(profileButton.href).toBe('http://localhost/profile');
}   );