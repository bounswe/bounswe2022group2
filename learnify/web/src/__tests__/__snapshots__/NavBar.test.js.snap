// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should render component layout correctly 1`] = `
<nav
  className="nav"
  data-testid="navbar"
>
  <a
    href="/home"
  >
    <img
      alt="Learnify Logo Navbar"
      height={60}
      src="logo-dblue.png"
    />
  </a>
  <ul>
    <div
      className="relative"
    >
      <input
        className="search-input-field"
        onChange={[Function]}
        placeholder="Search learning spaces"
        size={30}
        type="text"
        value=""
      />
      <div
        className="search-results-container"
      />
    </div>
    <li>
      <a
        className="navBarText"
        href="/notifications"
      >
        <img
          alt="Notifications Icon"
          height={20}
          src="notification-icon.svg"
        />
      </a>
    </li>
    <li>
      <a
        className="navBarText"
        href="/home"
      >
        Home
      </a>
    </li>
    <li>
      <a
        className="navBarText"
        href="/categories"
      >
        Categories
      </a>
    </li>
    <li>
      <a
        className="navBarText"
        href="/about"
      >
        About
      </a>
    </li>
    <li>
      <a
        className="navBarText"
        href="/contact"
      >
        Contact
      </a>
    </li>
    <li>
      <a
        className="navBarText"
        href="/profile"
      >
        Profile
      </a>
    </li>
  </ul>
</nav>
`;
